lib/cache/cache.go

```diff
@@ -269,16 +269,16 @@ type Cache struct {
 	// collections is a map of registered collections by resource Kind/SubKind
 	collections map[resourceKind]collection
 
-	trustCache         services.Trust
-	clusterConfigCache services.ClusterConfiguration
-	provisionerCache   services.Provisioner
-	usersCache         services.UsersService
-	accessCache        services.Access
+	trustCache         services.ServerTrust
+	clusterConfigCache services.ServerClusterConfiguration
+	provisionerCache   services.ServerProvisioner
+	usersCache         services.ServerUsers
+	accessCache        services.ServerAccess
 	dynamicAccessCache services.DynamicAccessExt
-	presenceCache      services.Presence
-	appSessionCache    services.AppSession
-	webSessionCache    types.WebSessionInterface
-	webTokenCache      types.WebTokenInterface
+	presenceCache      services.ServerPresence
+	appSessionCache    services.ServerAppSession
+	webSessionCache    services.ServerWebSessions
+	webTokenCache      services.ServerWebTokens
```

Comment:
shouldn't the `Cache` use client-facing interfaces?
its godoc says `Cache implements auth.AccessPoint interface and remembers the previously returned upstream value for each API call.`, which sounds like it should wrap a client